Loop/Managers/LoopDataManager.swift

```diff
@@ -970,7 +970,11 @@ extension LoopDataManager {
     ///     - LoopError.glucoseTooOld
     ///     - LoopError.pumpDataTooOld
     fileprivate func predictGlucose(
+        startingAt startingGlucose: GlucoseValue? = nil,
         using inputs: PredictionInputEffect,
+        historicalInsulinEffect insulinEffectOverride: [GlucoseEffect]? = nil,
+        insulinCounteractionEffects insulinCounteractionEffectsOverride: [GlucoseEffectVelocity]? = nil,
+        historicalCarbEffect carbEffectOverride: [GlucoseEffect]? = nil,
```

Comment:
While changing the bolus screen over the last few iterations, it's been a challenge to decouple LoopDataManager's use stored data (and effects derived from stored data) from the computations which combine them into a prediction or recommended bolus.

Adding API here to 'override' the stored effects with ones computed elsewhere isn't _super_ clean, but it's a minimal change, and significant refactoring in algorithm-land is high risk.

Reply:
It's a catch-22:
- Being able to refactor safely requires unit tests.
- Writing unit tests requires that code be sufficiently factored to break dependencies.

I'll see if I can pull together some tests here.